src/BladeRouteGenerator.php

```diff
@@ -21,16 +27,35 @@ public function getRoutePayload($group = false)
 
     public function generate($group = false)
     {
+        $this->prepareDomain();
+
         $json = $this->getRoutePayload($group)->toJson();
-        $appUrl = url('/') . '/';
+
         $routeFunction = file_get_contents(__DIR__ . '/js/route.js');
 
         return <<<EOT
 <script type="text/javascript">
     var namedRoutes = JSON.parse('$json'),
-        baseUrl = '$appUrl';
-        $routeFunction
+        baseUrl = '{$this->baseUrl}',
+        baseProtocol = '{$this->baseProtocol}',
+        baseDomain = '{$this->baseDomain}',
+        basePort = {$this->basePort};
+    $routeFunction
 </script>
 EOT;
     }
+
+    private function prepareDomain()
+    {
+        $url = url('/');
+        $this->baseUrl = $url . '/';
+        $this->baseProtocol = str_before($url, ':');
```

Comment:
I think this would break in the case where protocol was `//` and there was a port.
For example – `//example.com:8080`

In this case, `baseProtocol` would end up being `//example.com`

I think you could get around that by doing `str_before($url, '//')` and removing the `:` if it exists.

Reply:
can `url('/');` return a string like '//example.org:8080'? 

I think it takes the url from the request and so it should always contain a complete protocol and not the shortcut version.